From: Rob Herring <robh@kernel.org>
Subject: [PATCH] scripts/dtc: Fix multiple definition of `yylloc' with GCC 10+

In GCC 10+, -fno-common is the default. This causes linker errors due to
yylloc being defined in both dtc-parser.tab.o and dtc-lexer.lex.o.
Declare yylloc as extern in dtc-lexer.

Signed-off-by: Rob Herring <robh@kernel.org>
---
 scripts/dtc/dtc-lexer.l             | 2 +-
 scripts/dtc/dtc-lexer.lex.c_shipped | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/scripts/dtc/dtc-lexer.l b/scripts/dtc/dtc-lexer.l
--- a/scripts/dtc/dtc-lexer.l
+++ b/scripts/dtc/dtc-lexer.l
@@ -38,7 +38,7 @@ LINECOMMENT	"//".*\n
 #include "srcpos.h"
 #include "dtc-parser.tab.h"
 
-YYLTYPE yylloc;
+extern YYLTYPE yylloc;
 extern bool treesource_error;
 
 /* CAUTION: this will stop working if we ever use yyless() or yyunput() */
diff --git a/scripts/dtc/dtc-lexer.lex.c_shipped b/scripts/dtc/dtc-lexer.lex.c_shipped
--- a/scripts/dtc/dtc-lexer.lex.c_shipped
+++ b/scripts/dtc/dtc-lexer.lex.c_shipped
@@ -631,7 +631,7 @@ static int yy_init_globals (void );
 #include "srcpos.h"
 #include "dtc-parser.tab.h"
 
-YYLTYPE yylloc;
+extern YYLTYPE yylloc;
 extern bool treesource_error;
 
 /* CAUTION: this will stop working if we ever use yyless() or yyunput() */
-- 
2.25.1
